core/GlobalState.cc

```diff
@@ -2396,14 +2401,16 @@ void GlobalState::setPackagerOptions(const std::vector<std::string> &extraPackag
                                      const std::vector<std::string> &extraPackageFilesDirectorySlashPrefixes,
                                      const std::vector<std::string> &packageSkipRBIExportEnforcementDirs,
                                      const std::vector<std::string> &allowRelaxedPackagerChecksFor,
-                                     std::string errorHint) {
+                                     const std::vector<std::string> &packagerLayers, std::string errorHint) {
     ENFORCE_NO_TIMER(!packageDB_.frozen);
 
     packageDB_.enabled_ = true;
     packageDB_.extraPackageFilesDirectoryUnderscorePrefixes_ = extraPackageFilesDirectoryUnderscorePrefixes;
     packageDB_.extraPackageFilesDirectorySlashDeprecatedPrefixes_ = extraPackageFilesDirectorySlashDeprecatedPrefixes;
     packageDB_.extraPackageFilesDirectorySlashPrefixes_ = extraPackageFilesDirectorySlashPrefixes;
     packageDB_.skipRBIExportEnforcementDirs_ = packageSkipRBIExportEnforcementDirs;
+    absl::c_transform(packagerLayers, std::back_inserter(packageDB_.layers_),
+                      [this](string layer) { return enterNameUTF8(layer); });
```

Comment:
We can take this by `const string &` or `string_view`—`enterNameUTF8` does not take ownership of `layer`, so there's no need to copy.

Reply:
It's also legal to use `const auto&` here.